src/neuroconv/tools/nwb_helpers/_models/_hdf5_models.py

```diff
@@ -74,3 +75,18 @@ def get_data_io_keyword_arguments(self) -> Dict[str, Any]:
             compression_bundle = dict(compression=self.compression_method, compression_opts=self.compression_options)
 
         return dict(chunks=self.chunk_shape, **compression_bundle)
+
+
+class HDF5BackendConfiguration(BackendConfiguration):
+    """A model for matching collections of DatasetConfigurations specific to the HDF5 backend."""
+
+    backend: Literal["hdf5"] = Field(
+        default="hdf5", description="The name of the backend used to configure the NWBFile."
+    )
+    data_io_class: Type[H5DataIO] = Field(default=H5DataIO, description="The DataIO class that is specific to HDF5.")
```

Comment:
Similarly here, wouldn't HDF5 imply H5DataIO?

Reply:
Setting the `default` does set it as the value for that attribute; but again, users would never actually set this, it's mostly for internal convenience to have an object `<SomeBackend>BackendConfiguration` that reliably maps `.backend` as a short human readable string reference together with the actual `.data_io_class` which would be the actual thing you wrap the data data values in with `set_data_io`

Pretty much equivalent to having a dictionary `backend_to_data_io_class = dict(hdf5)` but a little more descriptive and extensible (primarily with respect to the list of contained DatasetConfigurations)